### PR TITLE
ci: update pipelines to run properly after move to nx

### PR DESCRIPTION
**Pull request by Hotell**
## Previous Behavior

- master pipelines are failing https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=354760&view=logs&j=f5e89eef-ad59-58c9-614d-c2fd6081e79d&t=7dd77012-753a-5f35-bf96-92957bb518f2

## New Behavior

- master pipelines are working

## Related Issue(s)

- Follows https://github.com/microsoft/fluentui/pull/31949

### CODE REVIEW

File: .devops/templates/build-test-lint.yml [hunk 1]
```diff
@@ -44,7 +44,13 @@ steps:
       DANGER_GITHUB_API_TOKEN: $(DANGER_GITHUB_API_TOKEN)
 
   - script: |
-      yarn buildci $(sinceArg)
```
Comment (Hotell) [inline .devops/templates/build-test-lint.yml:47]: in master this was running agains nothing basically `nx affected --base master --head master`

- it fails CI
- in order to mirror our existing behaviour ( run everything on master on every merge ) we need to use `run-many` => explicit API ✅

File: .devops/templates/deployE2E.yml [hunk 2]
```diff
@@ -18,18 +18,22 @@ steps:
...
   - script: |
       yarn nx affected -t build-storybook --nxBail $(sinceArg)
     displayName: build Storybooks
+    condition: eq(variables.isPR, true)
 
   ## This runs regardless of scope, the app will adapt to the scope as well
   - script: |
       yarn nx run pr-deploy-site:generate:site
     displayName: generate PR Deploy Site
+    condition: eq(variables.isPR, true)
```
Comment (Hotell) [inline .devops/templates/deployE2E.yml:32]: not needed on master pipeline

File: .devops/templates/deployE2E.yml [hunk 3]
```diff
@@ -18,18 +18,22 @@ steps:
...
 
   - task: AzureCLI@2
     displayName: Upload PR deploy site
+    condition: eq(variables.isPR, true)
```
Comment (Hotell) [inline .devops/templates/deployE2E.yml:36]: not needed on master pipeline

File: .devops/templates/deployE2E.yml [hunk 4]
```diff
@@ -45,6 +49,7 @@ steps:
 
   - task: GithubPRStatus@0
     displayName: 'Update PR deploy site github status'
+    condition: eq(variables.isPR, true)
```
Comment (Hotell) [inline .devops/templates/deployE2E.yml:52]: not needed on master pipeline

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Review by TristanWatanabe — APPROVED** [review]

**Pull request merged**